lib/Element/Base.php

```diff
@@ -50,7 +52,7 @@ public function __construct($value = null)
      *
      * If you are opening new elements, you must also close them again.
      */
-    public function xmlSerialize(Xml\Writer $writer)
+    public function xmlSerialize(Xml\Writer $writer): void
```

Comment:
IMO this sort of change is what forces us to make a major version bump to v3 - anything that currently has `Base.php` as its parent will notice this change. `xmlSerialize` "suddenly" has a return type restricted to `void`. A child that re-implements `xmlSerialize` is likely to not be specifying any return type. PHP will start complaining when it notices that the child "specifies" a wider return type than the parent, violating LSP.

The child will have to add `: void` - so we can't release this as a 2.x version, because then composer might give the new version to projects automagically before thir code is ready.

Reply:
Agree.

Alternatively do the PR on 2.x with all changes but these and do the BC breaking stuff separately

Reply:
> Alternatively do the PR on 2.x with all changes but these and do the BC breaking stuff separately

IMO the other parameter type changes... are "not that exciting" and mostly just in the test code. So there is not really much value for current users that are on PHP 7.1 7.2 7.3 to get those, they don't fix anything real.